app/src/main/java/com/github/se/polyfit/MainActivity.kt

```diff
@@ -86,8 +86,7 @@ class MainActivity : ComponentActivity() {
           composable(Route.PostInfo) { PostInfoScreen(navigation, navController) }
 
           composable(Route.CreatePost) {
-            CreatePostScreen(
-                navigation::goBack, navigation::navigateToHome, navigation::navigateToAddMeal)
+            CreatePostScreen(navigation::goBack, navigation::navigateToPostList)
```

Comment:
How come we're removing the navigateToAddMeal?

Reply:
Looking at the arguments of the function:

```kotlin
fun CreatePostScreen(
    navigateBack: () -> Unit = {},
    navigateForward: () -> Unit = {},
    navigateToAddMeal: () -> Unit = {},
    postViewModel: CreatePostViewModel = hiltViewModel()
)

``` 
The previous implementation takes the user to the overview screen instead of the add meal flow for some reason when the meal is empty. However, because the button is always disabled when the meal is empty, that functionality does not work. I think it makes more sense to take the user to the list of posts after creating a post rather than to the overview.

Reply:
Yeah I added it back. But it was originally in the spot of navigateForwards which caused some weird behavior.

Reply:
I updated my main and I guess that that is gone so it was fixed by another PR.